skfda/ml/regression/linear_model.py

```diff
@@ -0,0 +1,97 @@
+from sklearn.base import BaseEstimator, RegressorMixin
+from skfda.representation.basis import FDataBasis, Constant, Basis, FData
+
+import numpy as np
+
+
+class LinearScalarRegression(BaseEstimator, RegressorMixin):
+
+    def __init__(self, beta, weights=None):
+        self.beta_ = None
+        self.beta = beta
+        self.weights = weights
+
+    def fit(self, X, y):
+
+        y, X, beta, weights = self._argcheck(y, X)
+
+        nbeta = len(beta)
+        nsamples = X[0].nsamples
+
+        y = np.array(y).reshape((nsamples, 1))
+
+        for j in range(nbeta):
+            xcoef = X[j].coefficients
+            inner_basis_x_beta_j = X[j].basis.inner_product(beta[j])
+            inner_x_beta = (xcoef @ inner_basis_x_beta_j
+                            if j == 0
+                            else np.concatenate((inner_x_beta,
+                                                 xcoef @ inner_basis_x_beta_j),
+                                                axis=1))
+
+        if any(w != 1 for w in weights):
+            inner_x_beta = inner_x_beta * np.sqrt(weights)
+            y = y * np.sqrt(weights)
+
+        gram_inner_x_beta = inner_x_beta.T @ inner_x_beta
+        inner_x_beta_y = inner_x_beta.T @ y
+
+        gram_inner_x_beta_inv = np.linalg.inv(gram_inner_x_beta)
+        betacoefs = gram_inner_x_beta_inv @ inner_x_beta_y
+
+        idx = 0
+        for j in range(0, nbeta):
+            beta[j] = FDataBasis(beta[j], betacoefs[idx:idx+beta[j].nbasis].T)
+            idx = idx + beta[j].nbasis
+
+        self.beta_ = beta
+        return self
+
+    def predict(self, X):
+        return [sum(self.beta[i].inner_product(X[i][j])[0, 0] for i in
+                    range(len(self.beta))) for j in range(X[0].nsamples)]
+
+    def fit_predict(self, X, y):
```

Comment:
The default implementation already does that.

Reply:
There is not default implementation for BaseEstimator or RegressorMixin

![image](https://user-images.githubusercontent.com/2530727/60035742-da732a80-96ad-11e9-99b2-2c6dc7a997bb.png)

Reply:
I see... Maybe it is because predicting over the training data is useful for clustering, but not for regression. In that case I would still remove `fit_predict`.

Reply:
Perfect